src/utils/url.ts

```diff
@@ -11,3 +13,10 @@ export function buildSearchString(params: Record<string, string | undefined>): s
 
   return '?' + searchObj.toString()
 }
+
+export function replaceURL(url: string, strReplace: string, currentNetwork: number): string {
+  const re = strReplace ? /(([\w]+\/){0})([^/]+)(\/.+)*/gm : /(?:\/[^/]+){1}/
+  const subst = strReplace ? '$1' + strReplace + '$4' : ''
+  if (currentNetwork === Network.Mainnet) return url.replace(/^/, `/${strReplace}`)
+  else return url.replace(re, subst)
+}
```

Comment:
Please don't be angry with me hahaha

Just one last request.

Can you exit early when mainnet? It doesn't need the other parts and will look cleaner.
Something like:

```
  if (currentNetwork === Network.Mainnet) {
    return url.replace(/^/, `/${strReplace}`)
  }
  ... the rest here
```

After that you can merge it directly, you already got the approvals ;)